src/seismometer/data/otel.py

```diff
@@ -0,0 +1,694 @@
+}
+
+
+def do_auto_export(function_name: str, fn_settings: dict):
+    """Run a (metric-generating) function with
+    predetermined settings. To be used when reading in
+    an auto-generated config file, as opposed to a
+    manually-written one which takes a bit more
+    preprocessing.
+
+    Parameters
+    ----------
+    function_name : str
+        The name of the function to export.
+    fn_settings : dict
+        What settings (see output config) to apply:
+        args, kwargs: function parameters
+        extra_params: the current settings of Seismogram,
+        saved at the time of export.
+    """
+    from seismometer.api.plots import (
+        _plot_cohort_hist,
+        _plot_leadtime_enc,
+        model_evaluation,
+        plot_binary_classifier_metrics,
+        plot_cohort_evaluation,
+        plot_trend_intervention_outcome,
+    )
+    from seismometer.api.reports import feature_alerts, feature_summary, target_feature_summary
+    from seismometer.api.templates import show_cohort_summaries
+
+    args = fn_settings["args"]
+    kwargs = fn_settings["kwargs"]
+    # We need to have these here for circular import reasons.
+    match function_name:
+        case "feature_alerts":
+            fn = feature_alerts
+        case "feature_summary":
+            fn = feature_summary
+        case "model_evaluation":
+            fn = model_evaluation
+        case "plot_cohort_evaluation":
+            fn = plot_cohort_evaluation
+        case "plot_cohort_hist":
+            fn = _plot_cohort_hist
+        case "plot_leadtime_enc":
+            fn = _plot_leadtime_enc
+        case "plot_binary_classifier_metrics":
+            fn = plot_binary_classifier_metrics
+        case "plot_trend_intervention_outcome":
+            fn = plot_trend_intervention_outcome
+        case "show_cohort_summaries":
+            fn = show_cohort_summaries
+        case "target_feature_summary":
+            fn = target_feature_summary
+        case _:
+            raise ValueError(f"Unknown function name: {function_name}")
+    fn = allowed_export_names[function_name]
+    fn(*args, **kwargs)
+
+
+def extract_arguments(argument_names: list[str], run_settings: dict) -> dict:
+    """The YAML of a handwritten config file looks like this:
+    function_name:
+        cohorts: # ...
+        options:
+            # here's where we store the "extra information" per function call
+    So we get, out of options, the call parameters.
+
+    Parameters
+    ----------
+    argument_names : list[str]
+        Which arguments / other info (like Seismogram settings) we want to find.
+    run_settings : dict
+        The segment of the YAML we want to read.
+
+    Returns
+    -------
+    dict
+        The options and values we found in options.
+    """
+    return (
+        {arg: run_settings["options"][arg] for arg in argument_names if arg in run_settings["options"]}
+        if "options" in run_settings
+        else {}
+    )
```

Comment:
While valid python, it reads better with if blocks:

```suggestion
        if "options" in run_settings:
            return {arg: run_settings["options"][arg] for arg in argument_names if arg in run_settings["options"]}
        return {}
```

Reply:
This is true.